lib/public-stats.js

```diff
@@ -28,10 +44,48 @@ export const updatePublicStats = async ({ createPgClient, honestMeasurements })
       total = retrieval_stats.total + $1,
       successful = retrieval_stats.successful + $2
   `, [
-      total,
-      successful
+    total,
+    successful
+  ])
+}
+
+/**
+ * @param {import('./typings').CreatePgClient} createPgClient
+ * @param {string[]} participants
+ */
+const updateDailyParticipants = async (pgClient, participants) => {
+  debug('Updating daily participants (%s seen)', participants.length)
+  for (const participantAddress of participants) {
```

Comment:
Are we ok with performing 2 SQL queries for every participant, in series? If it fails midway, will we have inconsistent data? Playing devil's advocate here

Reply:
Great questions! 💯 

I'll think about this tomorrow.

Reply:
Yeah, I was focused so much on storage efficiency and the querying side, that I completely neglected the writing side.

With ~2k participants per round, my current implementation would run 4k SQL queries. That would take forever to complete.

Thanks for flagging this early! 😍

Reply:
Cool, I found a neat trick: We can use `SELECT UNNEST($1::TEXT[])` to build a query that accepts a JavaScript array parameter and converts the items into result rows.

Then I can change `INSERT...VALUES` to `INSERT...SELECT` to leverage this mechanism.

The changes were a bit more involved (see [7a2d14d](https://github.com/filecoin-station/spark-evaluate/pull/133/commits/7a2d14d11f10513e2946eba863f4721383ed322f)), but I expect the performance to be excellent.